src/validateAtoms.ts

```diff
@@ -0,0 +1,72 @@
+import { atom, SetStateAction, WritableAtom } from 'jotai';
+import { loadable } from 'jotai/utils';
+
+import type { CommonState } from './atomWithValidate';
+
+export type GetValues = <Value>(labeledAtoms: LabeledAtoms<Value>) => {
+  [k: string]: Value;
+};
+export type Validator = <Value>(values: { [k: string]: Value }) => Promise<any>;
+
+export type ValidatorState = {
```

Comment:
We want to return "values" too, no? I'm taking down one of my past ideas.

Reply:
We can return them, would make it consistent with the `atomWithValidate` API 

though, there's no actual use-case of the values here since we don't manage an atom with them.

Reply:
Like you said for a validator library, combined values can be useful for a certain UI library.
`waitForAll` is similar, in some cases we don't use the return value and in other cases we use it.

Reply:
I've made the change, 
just in case we extended on this with something else later

Reply:
Just have to write docs and examples of usage now

Reply:
> I've made the change

Not pushed yet??

Reply:
it’s in the same commit 
https://github.com/jotai-labs/jotai-form/pull/5/commits/2c02a50da9bb336474eee696725f768524247300

Reply:
okay, that's different from my expectation. I want it to be closer to waitForAll mental model (of mine...)

Reply:
oh, i’ll refer to the source of that and see what you mean